packages/core-database-sequelize/lib/connection.js

```diff
@@ -588,18 +588,14 @@ module.exports = class SequelizeConnection extends ConnectionInterface {
    * @return {[]String}
    */
   async getRecentBlockIds () {
-    if (!this.recentBlockIds.length) {
```

Comment:
This is a big hit on performance as already tested in the past. Constantly hitting the blocks table for a read operation is resource intense and can easily result in a deadlock.

Reply:
see my comment for @alexbarnsley this.recentBlockIds as no garantee for being in sync with the state of the blockchain

Reply:
Then `should be managed centrally at the level of the state machine` should be included in this PR as it is part of the same behaviour that is being modified.

Reply:
this more complex than that, opening an issue for that, see again te discussion